docs/extensions/index.md

```diff
@@ -13,7 +13,9 @@ This section covers how to write extensions.
 
 ## Extension Names
 
-All extension names follow the same convention as Java package names – they look like reversed domain names. (e.g.  `com.example.myextension`). For module-extensions, the last word in the module name will be the module's *short-name*. The short-name *must* be unique. It is possible to pick a different short-name, but that requires extra work.
+Pick a unique single word for your extension's name. Note that this name is used throughout the extension's code, e.g. function names so it should only include characters that can be safely used in that way.
+
+Nb. Historically, extension names used to follow the same convention as Java package names – they look like reversed domain names. (e.g.  `com.example.myextension`), but this is no longer the recommendation.
```

Comment:
```suggestion
``` info
    Historically, extension names used to follow the same convention as Java package names – they look like reversed domain names. (e.g.  `com.example.myextension`), but this is no longer the recommendation.
```